kartothek/core/dataset.py

```diff
@@ -381,16 +381,69 @@ def get_indices_as_dataframe(
         elif columns == []:
             return pd.DataFrame(index=self.partitions)
 
-        dfs = []
         columns_to_scan = columns[:]
         if predicates:
             predicate_columns = columns_in_predicates(predicates)
             # Don't use set logic to preserve order
+            # TODO why do I need to preserve the order here?
```

Comment:
That's a good question and doesn't make sense to me at all. If this is about deterministic behavior, a well-places `sorted` would also do the trick.

Reply:
I would assume this originated from a very early version. The only reason for not mixing the ordering I see here is determinism and this is only relevant for the output shape, i.e. we must preserve the original `columns` and reshape the output accordingly.

```python
columns_to_scan = sorted(set(columns) | set(columns_in_predicates))

# calc stuff

return df.loc[:, columns] # result with original columns

```

Reply:
Replaced it.